src/sagemaker/serve/builder/requirements_manager.py

```diff
@@ -0,0 +1,94 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""Requirements Manager class to pull in client dependencies from a .txt or .yml file"""
+from __future__ import absolute_import
+import logging
+import os
+import subprocess
+
+logger = logging.getLogger(__name__)
+
+
+class RequirementsManager:
+    """Manages dependency installation by detecting file types"""
+
+    def detect_file_exists(self, dependencies: str = None) -> str:
+        """Detects the type of file dependencies will be installed from
+
+        If a req.txt or conda.yml file is provided, it verifies their existence and
+        returns the local file path
+
+        Args:
+            dependencies (str): Local path where dependencies file exists.
+
+        Returns:
+            file path of the existing or generated dependencies file
+        """
+        dependencies = self._capture_from_local_runtime()
+
+        # Dependencies specified as either req.txt or conda_env.yml
+        if dependencies.endswith(".txt"):
+            self._install_requirements_txt()
```

Comment:
I think you're doing multiple things in one function. It's generally recommended to have functions as single tenant. Also, if you want to install while check path existence, you probably want to change the naming as well.

Reply:
My naming was making the functions confusing to understand, I will rename so they better fit their purpose. Thank you for pointing this out.